File: Leetcode/167. Two Sum II - Input Array Is Sorted/167_Two_Sum_2.py
from typing import List


class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        first, last = 0, len(numbers) - 1
        while first < last:
            sum = numbers[first] + numbers[last]
            if sum == target:
                return [first + 1, last + 1]
            elif sum < target:
                first += 1
            else:
                last -= 1
        return [-1, -1]


class Main:
    def main(self):
        s = Solution()
        numbers = [2, 7, 11, 15]
        target = 9
        print(s.twoSum(numbers, target))


if __name__ == "__main__":
    Main().main()

File: Leetcode/167. Two Sum II - Input Array Is Sorted/167_Two_Sum_2.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int first = 0, last = numbers.length - 1;
        int[] ans = new int[2];
        while (first < last) {
            int sum = numbers[first] + numbers[last];
            if (sum == target) {
                ans[0] = first + 1;
                ans[1] = last + 1;
                return ans;
            } else if (sum < target)
                first++;
            else if (sum > target)
                last--;
        }
        return ans;
    }

    public static void main(String[] args) {
        int[] numbers = { 2, 7, 11, 15 };
        Solution s = new Solution();
        int[] ans = s.twoSum(numbers, 9);
        System.out.println(ans[0] + " " + ans[1]);
    }
}